File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        res = 0
        currnt = 0
        
        vowels = "aeiou"

        for i in range(len(s)):
            if i >= k:          # 나가는 원소 확인
                if s[i-k] in vowels:
                    currnt -= 1
            
            if s[i] in vowels:  # i 가 커지기 전까지 윈도우를 확인
                currnt += 1
            
            res = max(currnt, res)
        return res

File: LeetCode 75/1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.cs
public class Solution {
    public int MaxVowels(string s, int k) {
        int res = 0;
        int currnt = 0;
        
        HashSet<char> vowels = new HashSet<char> { 'a', 'e', 'i', 'o', 'u' };

        for (int i = 0; i < s.Length; i++) {
            if (i >= k) {          // 나가는 원소 확인
                if (vowels.Contains(s[i-k])) {
                    currnt -= 1;
                }
            }
            
            if (vowels.Contains(s[i])) {  // i 가 커지기 전까지 윈도우를 확인
                currnt += 1;
            }
            
            res = Math.Max(currnt, res);
        }
        return res;
    }
}